File: java/src/main/java/api/jdy/WorkFlowDemo.java
package api.jdy;

import constants.HttpConstant;
import model.flow.WorkFlowApprovalCommentQueryParam;
import model.http.HttpRequestParam;
import org.apache.commons.lang3.StringUtils;
import util.HttpUtil;

import java.util.HashMap;
import java.util.Map;

import static constants.HttpConstant.APP_BASE_URL;

public class WorkFlowDemo {

    /**
     * 获取单条表单流程数据的审批意见
     *
     * @param queryParam
     * @throws Exception
     */
    public Map<String, Object> approvalComments(WorkFlowApprovalCommentQueryParam queryParam) throws Exception {
        if (StringUtils.isBlank(queryParam.getAppId()) || StringUtils.isBlank(queryParam.getEntryId())
                || StringUtils.isBlank(queryParam.getDataId())) {
            throw new RuntimeException("param lack!");
        }
        String url = APP_BASE_URL + queryParam.getAppId() + "/entry/" + queryParam.getEntryId()
                + "/data/" + queryParam.getDataId() + "/approval_comments";
        HttpRequestParam param = new HttpRequestParam();
        param.setApiKey(HttpConstant.API_KEY);
        param.setUrl(url);
        Map<String, Object> data = new HashMap<>();
        data.put("skip", queryParam.getSkip());
        data.put("limit", queryParam.getLimit());
        param.setData(data);
        Map<String, Object> result = HttpUtil.sendPostRequest(param);
        return result;
    }
}

File: python/src/api/jdy/work_flow_demo.py
from ...util import http_util as http_util
from ...constants.http_constant import HttpConstant
from ...model.http.http_request_param import HttpRequestParam

"""
获取单条表单流程数据的审批意见
Arguments:
    work_flow_query_param: WorkFlowApprovalCommentQueryParam 实例
"""


def approvalComments(param):
    url = HttpConstant.APPROVAL_COMMENTS_URL.format(appId=param.app_id, entryId=param.entry_id, dataId=param.data_id)
    print('url ====' + url)
    query_param = {'limit': param.limit, 'skip': param.skip}
    request_param = HttpRequestParam(HttpConstant.API_KEY, url, query_param)
    return http_util.send_post(request_param)
